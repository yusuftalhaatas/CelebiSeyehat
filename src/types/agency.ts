export type agency = {
  name: string;
  transport: Array<transport>;
  hotels: Array<hotel>;
};

export type transport = {
  name: string;
};

export type hotel = {
  name: string;
};
