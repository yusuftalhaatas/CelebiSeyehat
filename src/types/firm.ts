export type firm = {
  firmName: string;
  firmType: 'hotel' | 'transport';
  discountRate: 1 | 2;
  tickets?: {
    start: string;
    finish: string;
    price: number;
  };
};
