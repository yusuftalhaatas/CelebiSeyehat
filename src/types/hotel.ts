export type hotel = {
  firmName: string;
  city: string;
  firmType?: string;
  roomNo: string;
  date: string;
  price?: number;
  name: string;
  userName?: string;
};
