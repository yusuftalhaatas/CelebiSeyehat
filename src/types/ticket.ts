export type ticket = {
  date: string;
  start?: string;
  finish?: string;
  firmName: string;
  firmType?: 'hotel' | 'transport';
  no: string;
  userName?: string;
  name: string;
  price?: number;
};
