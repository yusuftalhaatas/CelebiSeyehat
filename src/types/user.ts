export type user = {
  name: string;
  id: string;
  puan: number;
  tickets: Array<ticket>;
};
export type ticket = {
  time: Date;
  no: number;
};
