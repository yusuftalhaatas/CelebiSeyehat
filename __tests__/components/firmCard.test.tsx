import React from 'react';
import {render} from '@testing-library/react-native';
import FirmCard from '../../src/components/firmCard';
import {firm} from '../../src/types/firm';

describe('components/FirmCard', () => {
  test('renders correctly', () => {
    const firm: firm = {
      firmName: 'Test Firm',
      firmType: 'transport',
      discountRate: 20,
    };

    const {getByText, toJSON} = render(<FirmCard {...firm} />);

    // snapshot test
    expect(toJSON()).toMatchSnapshot();

    // props test
    expect(getByText('Test Firm')).toBeDefined();
    expect(getByText('Discount:%20')).toBeDefined();
  });
});
