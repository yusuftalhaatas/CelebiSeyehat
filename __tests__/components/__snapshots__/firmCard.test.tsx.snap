// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FirmCard renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 15,
      "padding": 10,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Test Firm
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
        }
      }
    >
      Type:
      transport
    </Text>
    <Text
      style={
        {
          "fontSize": 20,
        }
      }
    >
      Discount:%
      20
    </Text>
  </View>
</View>
`;
